packages/core/src/modules/proofs/protocol/v1/V1ProofService.ts

```diff
@@ -996,6 +997,36 @@ export class V1ProofService extends ProofService<[IndyProofFormat]> {
     })
   }
 
+  public async getFormatData(
+    agentContext: AgentContext,
+    proofRecordId: string
+  ): Promise<GetFormatDataReturn<ProofFormat[]>> {
+    const [proposalMessage, requestMessage, presentationMessage] = await Promise.all([
+      this.findProposalMessage(agentContext, proofRecordId),
+      this.findRequestMessage(agentContext, proofRecordId),
+      this.findPresentationMessage(agentContext, proofRecordId),
+    ])
+
+    return {
+      proposalAttributes: proposalMessage?.presentationProposal?.attributes,
+      proposalPredicates: proposalMessage?.presentationProposal?.predicates,
+      proposal: proposalMessage
+        ? {
+            indy: proposalMessage,
```

Comment:
We should return the attachment data here rather than the messages itself. As you can see in the credentials implementation, we extract the attachment data: https://github.com/hyperledger/aries-framework-javascript/blob/main/packages/core/src/modules/credentials/protocol/v1/V1CredentialService.ts#L1085-L1116

Also, for the proposal message we should transform the message to an RFC0592 compliant proposal message, something like this, but for proofs: https://github.com/hyperledger/aries-framework-javascript/blob/main/packages/core/src/modules/credentials/protocol/v1/V1CredentialService.ts#L1085-L1087

Reply:
yes good spot. I will do that conversion in conjunction with the comment above.